packages/mds-api-server/index.ts

```diff
@@ -131,14 +131,16 @@ export const ApiServer = (
 }
 
 /* istanbul ignore next */
-export const checkAccess = (validator: (scopes: AccessTokenScope[]) => boolean) => (
-  req: ApiRequest,
-  res: ApiResponse,
-  next: express.NextFunction
-) => {
-  if (process.env.VERIFY_ACCESS_TOKEN_SCOPE === 'false' || validator(res.locals.scopes)) {
-    next()
-  } else {
-    res.status(403).send({ error: new AuthorizationError('no access without scope', { claims: res.locals.claims }) })
-  }
-}
+export const checkAccess = (validator: (scopes: AccessTokenScope[]) => boolean | Promise<boolean>) =>
```

Comment:
Coming back up to speed on my TS.  It looks like both return paths are async -- doesn't that imply that the return value is always `Promise<boolean>`?

Reply:
`checkAccess` itself actually returns an async middleware `(req, res, next)` function. 

The async middleware function is capable of using either a sync or async `validator` function, hence the `boolean | Promise<boolean>`

So, while the middleware is typically something simple like this:
```ts
checkAccess(scopes => scopes.includes('admin:all'))
```
The middleware could also use some more complex async validation (this is a contrived example for illustration)
```ts
checkAccess(async scopes => {
  // do some async validation
  const valid = await Promise.resolve(scopes.includes('admin:all'))
  return valid
})
```